### PR TITLE
chore(launch): Fix notebook tests that are broken in core

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

Modifies the tests for job creation in notebooks to directly check that a job was used on the run and that the name matches the expected structure.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

**Review by bcsherma — APPROVED** [review]

File: tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py [hunk 1]
```diff
@@ -259,21 +260,34 @@ def test_code_saving(notebook):
...
+
+    api = wandb.Api()
+    user = os.environ["WANDB_USERNAME"]
+    run = api.run(f"{user}/uncategorized/{run_id}")
+    used_artifacts = run.used_artifacts()
+    assert len(used_artifacts) == 1
+    assert (
+        used_artifacts[0].name
+        == "job-source-uncategorized-one_cell_disable_git.ipynb:v0"
+    )
```
Comment (kptkin) [inline tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:278]: do you think you can achieve something similar with the relay server? just wanted to check

File: tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py [hunk 2]
```diff
@@ -259,21 +260,34 @@ def test_code_saving(notebook):
...
 
-@pytest.mark.skip_wandb_core(feature="launch")
 def test_notebook_creates_artifact_job(notebook):
     with notebook("one_cell_disable_git.ipynb") as nb:
         nb.execute_all()
         output = nb.cell_output_html(2)
-        # 3 artifact files if code, 5 if also job
-        # TODO: test contents of artifact, requires relay server context for artifacts
-        assert "5 artifact file(s)" in output
+    regex_string = r'http:\/\/localhost:8080\/[^\/]+\/[^\/]+\/runs\/([^\'"]+)'
```
Comment (kptkin) [inline tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:267]: also adding a small comment about this regex would be helpful :)

File: tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py (hunk 1 above)
Comment (KyleGoyette) [reply tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:278]: I tried the relay server and afaict executing subprocesses within the context of the relay server doesn't work

Comment (kptkin) [reply tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:278]: oh, weird can you open an issue for us? i know @dmitryduev verified that pyrun works surprised about the subprocess :/

Comment (KyleGoyette) [reply tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:278]: My mistake, *notebooks* dont seem to hit the relay server, not sure about subprocesses.

Comment (kptkin) [reply tests/pytest_tests/system_tests/test_notebooks/test_notebooks.py:278]: super weird

**Pull request merged**